include/sys/zfs_refcount.h

```diff
@@ -114,6 +116,25 @@ typedef struct refcount {
 #define	zfs_refcount_count(rc) atomic_load_64(&(rc)->rc_count)
 #define	zfs_refcount_add(rc, holder) atomic_inc_64_nv(&(rc)->rc_count)
 #define	zfs_refcount_remove(rc, holder) atomic_dec_64_nv(&(rc)->rc_count)
+#define	zfs_refcount_remove_unless_last(rc, holder) ({		\
```

Comment:
I see equivalent in FreeBSD refcount(9) is called refcount_release_if_not_last().  Wonder if there is any wider consensus about the naming.
I also see plenty of random weird conditional atomics in Linux, wonder if any of them are reasonably better on any real-world architectures to bother using them.  I was told once that cmpxchg in a loop is not great.

Reply:
There is no better way than a cas loop on any architecture I heard of. Linux also rolls this way.